python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -89,13 +89,23 @@ def get_code_location_names(self) -> Iterable[str]:
 
 
 class CodeLocationDirectoryContext:
-    _components_registry: Mapping[str, RemoteComponentType] = {}
+    """Class encapsulating contextual information about a components code location directory.
+
+    Args:
+        root_path (Path): The absolute path to the root of the code location directory.
+        name (str): The name of the code location python package.
+        component_registry (ComponentRegistry): The component registry for the code location.
+        deployment_context (Optional[DeploymentDirectoryContext]): The deployment context containing
+            the code location directory. Defaults to None.
+        is_test_context (bool): If True, the code location will use the test component registry of
+            dagster-components instead of the the published component registry. Defaults to False.
+    """
 
```

Comment:
very skeptical of a boolean `is_test_context` flag here.

Should there just be test versions/test-approriate values for the other parameters?

Reply:
Good catch, that's actually some leftover detritus from v1 of this that I forgot to remove. Gone now.

The way it actually works is that the boolean gets passed to `from_path` where it gets used to resolve the `ComponentRegistry`.